### ISSUE TITLE
Flaky test or an bug when accessing the reload timestamp in the api after a config reload

### ISSUE THREAD

**Issue by ph** [elastic/logstash#5914]
https://logstash-ci.elastic.co/job/elastic+logstash+5.0+multijob-os-compatibility/141/os=fedora/console

```
Failures:

1) LogStash::Agent metrics after config reloading when reloading a good config does not set the failure reload timestamp
     Failure/Error: expect(value).to be(nil)
       
       expected nil
            got #<LogStash::Timestamp:476706> => 2016-09-14T02:03:39.152Z
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/fedora/logstash-core/spec/logstash/agent_spec.rb:437:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/fedora/vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/fedora/rakelib/test.rake:42:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/fedora/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/task.rb:248:in `execute'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/fedora/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/task.rb:243:in `execute'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/fedora/vendor/bundle/jruby/1.9/gems/rake

**Comment by ph** [comment]
Another related failure, maybe?

```

  1) LogStash::Agent metrics after config reloading when reloading a bad config increases the failed reload count
     Failure/Error: expect(value).to be > 1
       expected: > 1
            got:   1
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/logstash-core/spec/logstash/agent_spec.rb:504:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/rakelib/test.rake:42:in `(root)'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/task.rb:248:in `execute'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/task.rb:243:in `execute'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/task.rb:187:in `invoke_with_call_chain'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/task.rb:180:in `invoke_with_call_chain'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/task.rb:173:in `invoke'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/application.rb:152:in `invoke_task'
     # /var/lib/jenkins/workspace/elastic+logstash+5.0+multijob-os-compatibility/os/centos/vendor/bundle/jruby/1.9/gems/rake-11.2.2/lib/rake/application.rb:108:in `top_level'
```

https://logstash-ci.elastic.co/job/elastic+logstash+5.0+multijob-os-compatibility/140/os=centos/console

**Comment by ph** [comment]
Also https://logstash-ci.elastic.co/job/elastic+logstash+5.0+multijob-os-compatibility/140/os=debian/console

**Comment by jsvd** [comment]
should be fixed by #6039
